File: wrappers/javascript/anoncreds-shared/src/api/RevocationRegistry.ts
import type { JsonObject } from '../types'
import type { RevocationRegistryDefinition } from './RevocationRegistryDefinition'

import { AnoncredsObject } from '../AnoncredsObject'
import { anoncreds } from '../register'

export type RevokeCredentialOptions = {
  revocationRegistryDefinition: RevocationRegistryDefinition
  credentialRevocationIndex: number
  tailsPath: string
}

export type UpdateRevocationRegistryOptions = {
  revocationRegistryDefinition: RevocationRegistryDefinition
  issued: number[]
  revoked: number[]
  tailsDirectoryPath: string
}

export class RevocationRegistry extends AnoncredsObject {
  public static fromJson(json: JsonObject) {
    return new RevocationRegistry(anoncreds.revocationRegistryFromJson({ json: JSON.stringify(json) }).handle)
  }
}

File: wrappers/kotlin/src/commonMain/kotlin/api/RevocationRegistry.kt
package anoncreds.api

import anoncreds_rs.ObjectHandle
import anoncreds_rs.anoncreds_revocation_registry_from_json

class RevocationRegistry
private constructor(
    handle: ObjectHandle
): AnoncredsObject(handle) {
    companion object {
        fun fromJson(json: String): RevocationRegistry {
            return fromJson(
                json, ::RevocationRegistry,
                ::anoncreds_revocation_registry_from_json
            )
        }
    }
}
